boofuzz/sessions.py

```diff
@@ -1117,6 +1120,8 @@ def transmit_normal(self, sock, node, edge, callback_data):
                 self._fuzz_data_logger.log_fail(msg)
             else:
                 self._fuzz_data_logger.log_info(msg)
+        except (exception.BoofuzzTargetConnectionFailedError, exception.BoofuzzSSLError) as e:
+            self._fuzz_data_logger.log_fail(str(e))
```

Comment:
@SR4ven invalid/fuzzed payload should not cause an SSL error; however, I believe it should be treated as a test case failure. Given the code you referenced, I feel it might be worth giving the user the option to ignore SSL errors and trigger a restart of the target.

In all cases an SSL error would indicate that an EOT or socket-related error has occurred; however, the SSL API is not in a state where it has the complete encrypted data to complete the decryption. This could be caused by a connection loss before the expected encrypted data was received; or, a failed SSL-related operation (such as a session renegotiation).

In my particular case, the SSL error was thrown by ssl.recv() because my target threw a .Net stack exhaustion error. Given the particular SSL "reason" I got, it looks like the issue was the .Net SSL API ran of stack space in the process of encrypting the data that the target was attempting to send back.

Reply:
Thanks for the detailed info @kmendoza-bt.

Let's add a session option `ignore_connection_ssl_errors` to log SSL errors on info level so that the test case doesn't fail. Should be false by default.
You can take this code/option for reference:
https://github.com/jtpereyda/boofuzz/blob/c41860f30ed93cfb52a72bc8d103104c2451860e/boofuzz/sessions.py#L1085-L1090

Reply:
Fixed in commit b2b3cec